components/server/src/workspace/workspace-starter.ts

```diff
@@ -336,14 +336,14 @@ export class WorkspaceStarter {
 
             // check if there has been an instance before, i.e. if this is a restart
             const pastInstances = await this.workspaceDb.trace({ span }).findInstances(workspace.id);
-            const hasValidBackup = pastInstances.some(
-                (i) => !!i.status && !!i.status.conditions && !i.status.conditions.failed,
-            );
             let lastValidWorkspaceInstance: WorkspaceInstance | undefined;
-            if (hasValidBackup) {
-                lastValidWorkspaceInstance = pastInstances.reduce((previousValue, currentValue) =>
-                    currentValue.creationTime > previousValue.creationTime ? currentValue : previousValue,
-                );
+            // Sorted from latest to oldest
+            for (const i of pastInstances.sort((a, b) => (a.creationTime > b.creationTime ? -1 : 1))) {
```

Comment:
Would we want to sort on `stopped` time? A backup takes place after a workspace stops which means that we're looking for stopped workspaces, not creation time, no?

Reply:
Instances are totally ordered, so it should not make a difference, except that creationTime might be a little less brittle.

Reply:
Ah, it's an instance, not a workspace. Thanks.